lib/client/rpc/proposals.ts

```diff
@@ -242,7 +251,184 @@ export class FutarchyRPCProposalsClient implements FutarchyProposalsClient {
+    const proposalKP = Keypair.generate();
+    const proposal = proposalKP.publicKey;
+
+    console.log(proposal)
+    const currentDao = daoAggregate.daos.filter((dao) => dao.protocol.autocrat.programId.toString() == this.autocratClient.autocrat.programId.toString())[1]
+    if (!currentDao) return
+
+    console.log(currentDao.publicKey.toString())
+    const autocrat = this.autocratClient.autocrat
+    const vaultClient = this.autocratClient.vaultClient
+    const ammClient = this.autocratClient.ammClient
+    const treasury = currentDao.daoAccount.treasury
+
+    const usdcMint = currentDao.daoAccount.usdcMint;
+    const tokenMint = currentDao.daoAccount.tokenMint
+    if (!tokenMint) return
+    const vaultProgramId = vaultClient.vaultProgram.programId
+
+    console.log(usdcMint.toString(), tokenMint.toString())
+    const [baseVault] = getVaultAddr(vaultProgramId, treasury, tokenMint, proposal);
+    const [quoteVault] = getVaultAddr(vaultProgramId, treasury, usdcMint, proposal);
+
+    const [passBase] = getVaultFinalizeMintAddr(vaultProgramId, baseVault);
+    const [passQuote] = getVaultFinalizeMintAddr(vaultProgramId, quoteVault);
+    const [passAmm] = getAmmAddr(ammClient.program.programId, passBase, passQuote, proposal);
+
+    const [failBase] = getVaultRevertMintAddr(vaultProgramId, baseVault);
+    const [failQuote] = getVaultRevertMintAddr(vaultProgramId, quoteVault);
+    const [failAmm] = getAmmAddr(ammClient.program.programId, failBase, failQuote, proposal);
+
+    const daoAccount = await autocrat.account.dao.fetch(currentDao.publicKey)
+
+    const baseTokensToLP: BN = daoAccount.minBaseFutarchicLiquidity
+    const quoteTokensToLP: BN = daoAccount.minQuoteFutarchicLiquidity
+
+
+    console.log(baseTokensToLP.toNumber(), quoteTokensToLP.toNumber())
+    const baseBalance = await this.getUserTokenBalance(tokenMint)
+    const quoteBalance = await this.getUserTokenBalance(usdcMint)
+    console.log(baseBalance, quoteBalance)
+    const memoIx = this.createMemoInstruction()
+
+    const initializeBaseVaultIx = await this.autocratClient.vaultClient.initializeVaultIx(treasury, tokenMint, proposal).transaction()
+    const initializeQuoteIx = await this.autocratClient.vaultClient.initializeVaultIx(treasury, usdcMint, proposal).transaction()
+
+    console.log(tokenMint.toString())
+
+
+
+
+    const mintBaseTokensTx = await vaultClient.mintConditionalTokensIx(baseVault, tokenMint, baseTokensToLP, user).transaction()
+    const mintQuoteTokensTx = await vaultClient.mintConditionalTokensIx(quoteVault, usdcMint, quoteTokensToLP, user).transaction()
+
+    const createAmmdAddLiquidityTx = await ammClient
+      .createAmmIx(passBase, passQuote, daoAccount.twapInitialObservation, daoAccount.twapMaxObservationChangePerUpdate, proposal)
+      .postInstructions([
+        await ammClient.createAmmIx(failBase, failQuote, daoAccount.twapInitialObservation, daoAccount.twapMaxObservationChangePerUpdate, proposal).instruction(),
```

Comment:
I think reviewing this is called for where you confirm you've got the IXs and just place them in variables so it's easier to understand the downstream effects.